test/hotspot/jtreg/runtime/Thread/InterruptAtExit.java

```diff
@@ -42,32 +42,46 @@ public void run() {
         // Tell main thread we have started.
         startSyncObj.countDown();
         try {
-            // Wait for main thread to interrupt us so we
-            // can race to exit.
+            // Wait for main thread to tell us to race to the exit.
             exitSyncObj.await();
         } catch (InterruptedException e) {
-            // ignore because we expect one
+            // Ignore because we are testing java.lang.Thread.interrupt()
+            // and one may arrive before we leave the 'try { }' block.
         }
     }
 
     public static void main(String[] args) {
-        InterruptAtExit threads[] = new InterruptAtExit[N_THREADS];
+        int timeMax = 0;
+        if (args.length == 0) {
+            timeMax = DEF_TIME_MAX;
+        } else {
+            try {
+                timeMax = Integer.parseUnsignedInt(args[0]);
+            } catch (NumberFormatException nfe) {
+                System.err.println("'" + args[0] + "': invalid timeMax value.");
+                    usage();
+            }
+        }
 
-        for (int i = 0; i < N_THREADS; i++ ) {
-            threads[i] = new InterruptAtExit();
-            int late_count = 1;
-            threads[i].start();
+        System.out.println("About to execute for " + timeMax + " seconds.");
+
+        long count = 0;
+        long start_time = System.currentTimeMillis();
+        while (System.currentTimeMillis() < start_time + (timeMax * 1000)) {
+            count++;
+
+            InterruptAtExit thread = new InterruptAtExit();
+            thread.start();
             try {
                 // Wait for the worker thread to get going.
-                threads[i].startSyncObj.await();
-
-                // The first interrupt() call will break the
-                // worker out of the exitSyncObj.await() call
-                // and the rest will come in during thread exit.
-                for (; late_count <= N_LATE_CALLS; late_count++) {
-                    threads[i].interrupt();
+                thread.startSyncObj.await();
+                // Tell the worker thread to race to the exit and the
+                // Thread.interrupt() calls will come in during thread exit.
+                thread.exitSyncObj.countDown();
+                while (true) {
+                    thread.interrupt();
```

Comment:
I just want to make sure I correctly understand the intent here. You call countDown() to unblock the thread's run() method, and then immediately interrupt the thread to create a race between the thread doing a normal exit and the thread being interrupted before or as it exits.

Reply:
That is correct. The intention is to stress the Thread.interrupt() calls
on the exiting thread.